evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -72,9 +72,15 @@ where
     /// The aggregation circuit, which verifies two proofs that can either be
     /// root or aggregation proofs.
     pub aggregation: AggregationCircuitData<F, C, D>,
+    /// The two-to-one aggregation circuit, which verifies two unrelated
+    /// aggregation proofs.
+    pub two_to_one_aggregation: TwoToOneAggCircuitData<F, C, D>,
     /// The block circuit, which verifies an aggregation root proof and an
     /// optional previous block proof.
     pub block: BlockCircuitData<F, C, D>,
+    /// The two-to-one block aggregation circuit, which verifies two unrelated
+    /// block proofs.
+    pub two_to_one_block: TwoToOneAggCircuitData<F, C, D>,
```

Comment:
I'm a bit confused about the `two_to_one_aggregation`.
I see the use of aggregating two final block proofs together, but why aggregating intermediary layers that don't share anything?

Reply:
I think it could be useful in some cases. A "block proof" is just an "aggregation proof" along with a recursively verified previous block proof. But in some cases, the previous batch is already verified elsewhere (e.g. on L1), so the recursive proof is redundant.
I might be missing something though.